w3af/plugins/grep/private_keys.py

```diff
@@ -0,0 +1,98 @@
+"""
+private_keys.py
+
+Copyright 2006 Andres Riancho
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+
+"""
+import re
+
+import w3af.core.data.constants.severity as severity
+
+from w3af.core.controllers.plugins.grep_plugin import GrepPlugin
+from w3af.core.data.kb.vuln import Vuln
+
+
+class private_keys(GrepPlugin):
+    """
+    Grep every page for private keys.
+
+    :author: Yvonne Kim
+    """
+    def __init__(self):
+        GrepPlugin.__init__(self)
+
+        key_regex = '-----BEGIN RSA PRIVATE KEY-----\n[^-]+\n-----END RSA PRIVATE KEY-----'
```

Comment:
It would be nice to have regular expressions for the most common private key formats, and also, maybe, for the public key formats too.

When adding those it might make sense to `MultiRe` (matches multiple regular expressions against one response, all at the same time, faster than a for loop). You can find examples on how that is used in other `grep` plugins.

Also, I'm not sure about the regular expression you used:

```
-----BEGIN RSA PRIVATE KEY-----\n[^-]+\n-----END RSA PRIVATE KEY-----
```

Maybe it is easier and more effective to search for `-----BEGIN RSA PRIVATE KEY-----`? Matching new lines inside regular expressions might be tricky.

Reply:
I plan on adding more regexes for other formats next. 

Searching for just the header would work, but I'm wary about that because it doesn't necessarily indicate that a key is present. I'll take the newline matching out.

Reply:
Oh, if I were to search for public key formats as well, it would be trivial- replacing instances of "PRIVATE" in the headers with "PUBLIC". Although, what would the value of searching for public keys be if they're freely available? Thanks for your patience!

Reply:
Well, a "common site" (not stackoverflow or developer forum) which has a public key somewhere is very strange and should be manually reviewed. Maybe w3af can find the public key somewhere, but not the private key. Then the w3af user can use his experience and knowledge about the site to find the private key.